File: solutions/typescript/darts/1/darts.ts
export function score(x: number, y: number): number {
  const rs = x**2 + y**2;
  if (rs > 100) {
    return 0
  } else if (rs > 25) {
    return 1
  } else if (rs > 1) {
    return 5
  } else {
    return 10
  }
}

File: solutions/zig/darts/1/darts.zig
pub const Coordinate = struct {
    // This struct, as well as its fields and methods, needs to be implemented.

    r2: f32,

    pub fn init(x: f32, y: f32) Coordinate {
        return Coordinate{ .r2 = x * x + y * y };
    }
    pub fn score(self: Coordinate) usize {
        if (self.r2 > 100) {
            return 0;
        } else if (self.r2 > 25) {
            return 1;
        } else if (self.r2 > 1) {
            return 5;
        } else {
            return 10;
        }
    }
};
